tests/activity_api_rules_test.go

```diff
@@ -0,0 +1,260 @@
+		return err
+	}
+}
+
+func (s *ActivityApiRulesClientTestSuite) createWorkflow(ctx context.Context, workflowFn WorkflowFunction) sdkclient.WorkflowRun {
+	workflowOptions := sdkclient.StartWorkflowOptions{
+		ID:        testcore.RandomizeStr("wf_id-" + s.T().Name()),
+		TaskQueue: s.TaskQueue(),
+	}
+	workflowRun, err := s.SdkClient().ExecuteWorkflow(ctx, workflowOptions, workflowFn)
+	s.NoError(err)
+	s.NotNil(workflowRun)
+
+	return workflowRun
+}
+
+func (s *ActivityApiRulesClientTestSuite) TestActivityRulesApi_WhileRetrying() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	testWorkflow := newInternalRulesTestWorkflow(ctx, &s.FunctionalTestBase)
+
+	s.Worker().RegisterWorkflow(testWorkflow.WorkflowFunc)
+	s.Worker().RegisterActivity(testWorkflow.ActivityFunc)
+
+	workflowRun := s.createWorkflow(ctx, testWorkflow.WorkflowFunc)
+
+	// wait for activity to start and fail few times
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		if description.GetPendingActivities() != nil {
+			assert.Len(t, description.PendingActivities, 1)
+		}
+		assert.Less(t, int32(1), testWorkflow.startedActivityCount.Load())
+	}, 10*time.Second, 200*time.Millisecond)
+
+	// create rule to pause activity
+	ruleID := "pause-activity"
+	activityType := "ActivityFunc"
+	createRuleRequest := &workflowservice.CreateWorkflowRuleRequest{
+		Namespace: s.Namespace().String(),
+		Spec: &rulespb.WorkflowRuleSpec{
+			Id: ruleID,
+			Trigger: &rulespb.WorkflowRuleSpec_ActivityStart{
+				ActivityStart: &rulespb.WorkflowRuleSpec_ActivityStartTrigger{
+					Predicate: fmt.Sprintf("ActivityType = \"%s\"", activityType),
+				},
+			},
+			Actions: []*rulespb.Action{
+				{
+					Variant: &rulespb.Action_Pause{
+						Pause: &rulespb.Action_ActionPause{
+							Scope: enumspb.RULE_ACTION_SCOPE_ACTIVITY,
+						},
+					},
+				},
+			},
+		},
+	}
+	createRuleResponse, err := s.FrontendClient().CreateWorkflowRule(ctx, createRuleRequest)
+	s.NoError(err)
+	s.NotNil(createRuleResponse)
+
+	// verify that frontend has updated namespaces
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		nsResp, err := s.FrontendClient().ListWorkflowRules(ctx, &workflowservice.ListWorkflowRulesRequest{
+			Namespace: s.Namespace().String(),
+		})
+		s.NoError(err)
+		s.NotNil(nsResp)
+		s.NotNil(nsResp.Rules)
+		if nsResp.GetRules() != nil {
+			s.Len(nsResp.Rules, 1)
+			s.Equal(ruleID, nsResp.Rules[0].Spec.Id)
+		}
+	}, 5*time.Second, 200*time.Millisecond)
+
+	// wait for activity to be paused by rule
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		description, err := s.SdkClient().DescribeWorkflowExecution(ctx, workflowRun.GetID(), workflowRun.GetRunID())
+		assert.NoError(t, err)
+		if description.GetPendingActivities() != nil {
+			assert.Len(t, description.PendingActivities, 1)
```

Comment:
Like here. I needs to be:
```suggestion
			if(assert.Len(t, description.PendingActivities, 1)) { ...
```